### PR TITLE
Feature sc linkage

### PR DESCRIPTION
**Pull request by holke**
**_Describe your changes here:_**

The compiled t8 libraries now link correctly against p4est and sc.
See also #224 

**_All these boxes must be checked by the reviewers before merging the pull request:_**

- [x] The author added a BSD statement to `doc/` (or already has one)
- [x] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)

  If the Pull request indroduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

- [x] All tests pass (in various configurations, this should be executed automatically in a github action)
- [x] New source/header files are properly added to the Makefiles
- [x] The reviewer executed the new code features at least once and checked the results manually
- [x] The code is covered in an existing or new test case
- [x] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [x] The code is well documented
- [x] All function declarations, structs/classes and their members have a proper doxygen documentation
- [x] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)
- [x] Testing of this template: If you feel something is missing from this list, contact the developers

### CODE REVIEW

**Comment by holke** [comment]
nm still lists the p4est and sc functions as `U`, so please check whether this really fixes the issue.

**Review by jmark — COMMENTED** [review]
Nice!

**Review by jmark — APPROVED** [review]

File: src/Makefile.am [hunk 1]
```diff
@@ -73,6 +73,7 @@ src_libt8_la_SOURCES = \
 	$(libt8_compiled_sources)
 src_libt8_la_CPPFLAGS = $(AM_CPPFLAGS) $(T8_CPPFLAGS)
 src_libt8_la_LDFLAGS = -release $(VERSION)
+src_libt8_la_LIBADD = @T8_P4EST_LDADD@ @T8_SC_LDADD@
```
Comment (holke) [inline src/Makefile.am:76]: # comment to trigger github action

Comment (holke) [inline src/Makefile.am:76]:
```suggestion
src_libt8_la_LIBADD = @T8_P4EST_LDADD@ @T8_SC_LDADD@
#comment to trigger github action
```

File: src/Makefile.am [hunk 2]
```diff
@@ -73,6 +73,8 @@ src_libt8_la_SOURCES = \
...
+#comment to trigger github action
```
Comment (holke) [inline src/Makefile.am:77]:
```suggestion
```

**Pull request merged**